tsl/src/planner.c

```diff
@@ -146,3 +149,82 @@ tsl_set_rel_pathlist(PlannerInfo *root, RelOptInfo *rel, Index rti, RangeTblEntr
 
 	ts_cache_release(hcache);
 }
+
+/*
+ * Check session variable to disable distributed insert with COPY.
+ *
+ * Allows testing and comparing different insert plans. Not a full GUC, since
+ * in most cases this is not something that users are expected to disable.
+ */
+static bool
+copy_mode_enabled(void)
+{
+	const char *enable_copy =
+		GetConfigOption("timescaledb.enable_distributed_insert_with_copy", true, false);
```

Comment:
I think this should be real GUC, all the other planner switches are real GUCs and it would give proper datatype support. Most of those planner GUCs are not expected to be changed by users during normal operations but they are a useful tool for support in  case of bugs and having them be a real GUC makes them more convenient as they are discoverable.

Reply:
I think the value of a GUC is limited for the user and leaves us less flexibility to change the INSERT path in the future. This is not a `SELECT`/query path where you'd like to pick one node over the other depending on how you'd like to optimize your query. Instead, I'd prefer the option to continue to evolve and experiment with the INSERT path, potentially replacing this node with something else in the future (given various caveats, etc). If we expose a GUC, then that's an "official" settings variable that users might come to rely on and it's difficult to remove or change.

This can always be elevated to GUC in the future.